datafusion/optimizer/src/type_coercion.rs

```diff
@@ -119,6 +134,41 @@ impl ExprRewriter for TypeCoercionRewriter {
 
     fn mutate(&mut self, expr: Expr) -> Result<Expr> {
         match expr {
+            Expr::ScalarSubquery(Subquery { subquery }) => {
+                let mut optimizer_config = OptimizerConfig::new();
```

Comment:
I think we should pass through the same OptimizerConfig (so it has the same context needed to evaluate `now()` rather than creating new ones

Reply:
After we move the `type coercion` out of the optimizer framework, we don't need the parameter of `OptimizerConfig` to dfs the plan when doing the type coercion.

I think the `new` of `OptimizerConfig` will not affect the rule of `type coercion`, because the rule of `type coercion` don't use the `OptimizerConfig` to generate some values.

Reply:
I think when we complete this issue https://github.com/apache/arrow-datafusion/issues/3582, this parameter will be dropped.
We don't need to concern this.